src/sinks/aws_cloudwatch_metrics.rs

```diff
@@ -0,0 +1,381 @@
+                    metric_name: "temperature".into(),
+                    value: Some(10.0),
+                    ..Default::default()
+                }],
+            }
+        );
+    }
+
+    #[test]
+    fn encode_events_histogram() {
+        let namespace = String::from("namespace");
+
+        let events = vec![Event::Metric(Metric::Histogram {
+            name: "latency".into(),
+            val: 11.0,
+            sample_rate: 100,
+        })];
+
+        assert_eq!(
+            encode_events(events, namespace.clone()).unwrap(),
+            PutMetricDataInput {
+                namespace,
+                metric_data: vec![MetricDatum {
+                    metric_name: "latency".into(),
+                    values: Some(vec![11.0]),
+                    counts: Some(vec![100.0]),
+                    ..Default::default()
+                }],
+            }
+        );
+    }
+}
+
+#[cfg(feature = "cloudwatch-metrics-integration-tests")]
+#[cfg(test)]
+mod integration_tests {
+    use super::*;
+    use crate::region::RegionOrEndpoint;
+    use crate::test_util::random_string;
+    use futures::{stream, Sink};
+    use tokio::runtime::Runtime;
+
+    fn config() -> CloudWatchMetricsSinkConfig {
+        CloudWatchMetricsSinkConfig {
+            namespace: "vector".into(),
+            region: RegionOrEndpoint::with_endpoint("http://localhost:4582".to_owned()),
+            ..Default::default()
+        }
+    }
+
+    #[test]
+    fn cloudwatch_metrics_healthchecks() {
+        let mut rt = tokio::runtime::Runtime::new().unwrap();
+
+        let healthcheck = CloudWatchMetricsService::healthcheck(&config()).unwrap();
+        rt.block_on(healthcheck).unwrap();
+    }
+
+    #[test]
+    fn cloudwatch_metrics_put_data() {
+        let mut rt = Runtime::new().unwrap();
+        let sink = CloudWatchMetricsService::new(config(), Acker::Null).unwrap();
+
+        let mut events = Vec::new();
+
+        let counter_name = random_string(10);
+        for i in 0..10 {
+            let event = Event::Metric(Metric::Counter {
+                name: counter_name.clone(),
+                val: i as f32,
+            });
+            events.push(event);
+        }
+
+        let gauge_name = random_string(10);
+        for i in 0..10 {
+            let event = Event::Metric(Metric::Gauge {
+                name: gauge_name.clone(),
+                val: i as f32,
+                direction: None,
+            });
+            events.push(event);
+        }
+
+        let histogram_name = random_string(10);
+        for i in 0..10 {
+            let event = Event::Metric(Metric::Histogram {
+                name: histogram_name.clone(),
+                val: i as f32,
+                sample_rate: 100,
+            });
+            events.push(event);
+        }
+
+        let stream = stream::iter_ok(events.clone().into_iter());
+
+        let pump = sink.send_all(stream);
+        rt.block_on(pump).unwrap();
```

Comment:
is there any way we can check that these items were inserted correctly?

Reply:
I've tried to find a working combination of localstack/moto, rusoto and friends to make this work, but I ran out of time unfortunately.

- list_metrics was not working out of the box
- get_metric_statistics either
- get_metric_data is not yet implemented https://github.com/spulec/moto/issues/2239

Not sure if I should skip this part of test for now, or spend some more time to make it work.

Reply:
So pretty much there is no way to fetch metrics? 😢

Reply:
I haven't found it yet 🤷‍♂

Reply:
Time to open an angry issue with localstack 😂 

Anyways that is fine, though ideally we should be checking but I'm not sure how to accomplish this without us making a PR to localstack.